napari/layers/shapes/_shapes_mouse_bindings.py

```diff
@@ -211,7 +286,10 @@ def vertex_insert(layer, event):
         return
 
     # Determine the closet edge to the current cursor coordinate
-    ind, loc = point_to_lines(layer.displayed_coordinates, all_edges)
+    displayed_coordinates = [
+        cursor_event.data_position[i] for i in layer._dims_displayed
+    ]
```

Comment:
this seem to be a frequently used idiom. Maybe a method on data_position if me make it a named tuple or similar ?

Reply:
This is something I want to fix and remove soon, maybe I should add a comment indicating that as it isn't clear. One day I hope `layer._dims_displayed` will no longer exist :-)

Reply:
This has now been removed, thanks to #2327